blob/s3blob/s3blob.go

```diff
@@ -134,19 +134,25 @@ func (o *URLOpener) OpenBucketURL(ctx context.Context, u *url.URL) (*blob.Bucket
 }
 
 // Options sets options for constructing a *blob.Bucket backed by fileblob.
-type Options struct{}
+type Options struct {
+	UseLegacyListObjects bool
```

Comment:
Please add a docstring for the new option. Perhaps explain that some providers have S3-compatible APIs, but don't support the ListObjectV2 API yet.